blobmodel/stochasticality.py

```diff
@@ -0,0 +1,114 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from .blobs import Blob
+from nptyping import NDArray, Float
+from typing import Any
+
+
+class BlobFactory(ABC):
+    """
+    Abstract class used by 2d propagating blob model to specify blob parameters. 
+    """
+
+    @abstractmethod
+    def sample_blobs(
+        self, Ly: float, T: float, num_blobs: int, blob_shape: str, t_drain: float
+    ) -> list[Blob]:
+        raise NotImplementedError
+
+
+class DefaultBlobFactory(BlobFactory):
+    """
+    Defualt blob parameters 
+    """
+
+    def __init__(
+        self,
+        A_dist: str = "exp",
+        W_dist: str = "exp",
+        vx_dist: str = "deg",
+        vy_dist: str = "normal",
+        A_parameter: float = 1.0,
+        W_parameter: float = 1.0,
+        vx_parameter: float = 1.0,
+        vy_parameter: float = 1.0,
+    ) -> None:
+        self.A_dist = A_dist
+        self.W_dist = W_dist
+        self.vx_dist = vx_dist
+        self.vy_dist = vy_dist
+        self.A_parameter = A_parameter
+        self.W_parameter = W_parameter
+        self.vx_parameter = vx_parameter
+        self.vy_parameter = vy_parameter
+
+    def __choose_distribution(
+        self, dist_type: str, free_parameter: float, num_blobs: int,
+    ) -> NDArray[Any, Float[64]]:
```

Comment:
Minor: I think the name of this private method is a bit misleading. The method is not choosing a distribution, instead it is drawing random variables from a given distribution. I would call it draw_random_variables, or something like that